### PR TITLE
Remove userids

### LINKED ISSUE #356: Remove user ids
I'm done. We need to get rid of the user id. It shouldn't exist.

List of what needs to be done:

1. Change all calls to users to use name rather than id  
2. Change creation of users  
3. Create script to remove ids, and change all references/games/etc to use username
4. Test script works

And during deployment:

1. Take down fapp  
2. Backup, move offsite
2. Deploy new code  
3. Run script  
4. Ensure script ran successfully
5. Bring fapp back up

### PR DESCRIPTION
**Pull request by not-an-aardvark**
Resolves #356. Also see the database migration script [here](https://gist.github.com/not-an-aardvark/7473682691b33ae879e1).

### CODE REVIEW

**Comment by YaManicKill** [comment]
Looks good in general. I'll pull it down this evening to see if I can see if anything is missing.

We'll obviously need to run this in beta, but to do that we'll need to duplicate the database, methinks. 

I think if we duplicate the prod database temporarily, run the script, deploy the version and then we can test to make sure that it seems good.

Is there anything missing that you know of from this, or should it all be done?

**Comment by not-an-aardvark** [comment]
Everything is done as far as I can tell.

**Comment by YaManicKill** [comment]
This'll need rebased as well.

**Comment by not-an-aardvark** [comment]
Rebased. This branch is up-to-date with the `version-2.2` branch as of now.

**Comment by not-an-aardvark** [comment]
I refactored a bunch of clientside stuff. The only front-end difference so far is that the apply-for-flair window now looks like [this](https://i.gyazo.com/17734912faadd7da4989feea0f847c1d.png) (though the flairs themselves will be normal when we deploy it).

File: assets/js/refCtrl.js [hunk 1]
```diff
@@ -375,27 +237,10 @@ module.exports = function ($scope, $filter) {
...
-    var url = "/mod/setlocalban";
-    io.socket.post(url, {userId: $scope.refUser.id, ban: ban}, function (data, res) {
-      if (res.statusCode === 200) {
-        $scope.refUser.banned = data.banned;
-        $scope.$apply();
-      } else {
-        console.log("Error");
-      }
-    });
+    return sharedService.setLocalBan($scope, $scope.refUser.name, ban);
```
Comment (YaManicKill) [inline assets/js/refCtrl.js]: We shouldn't need this one in multiple places, anyway.

Comment (not-an-aardvark) [inline assets/js/refCtrl.js]: One of them is for the ban/unban button on user profiles, and one of them is for the unban button on the banlist.

That said, I'm not sure we need the one on the banlist; I'mean occasionally worried that I'll accidentally hit the wrong button and unban someone.

**Comment by not-an-aardvark** [comment]
Closing since this is out of date -- see [the new PR](https://github.com/YaManicKill/flairhq/pull/387).

**Pull request closed without merging**